File: joonas/069.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    // solve problem in [l, r)
    TreeNode* makeTree(vector<int>& a, int l, int r) {
        if (l >= r) return nullptr;
        int mid = (l + r) >> 1;
        auto left = makeTree(a, l, mid);
        auto right = makeTree(a, mid + 1, r);
        return new TreeNode(a[mid], left, right);
    }
    
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return makeTree(nums, 0, nums.size());
    }
};

File: jlogkim/Season1/069.java
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { 
        this.val = val; 
        left=null;
        right=null;}
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}
class Solution {
  TreeNode root = null;
  int mnums[];
  public TreeNode getSubTreeRoot(int st,int ed){
      int length=ed-st+1;
      int mid = st+length/2;
      TreeNode l=null;
      TreeNode r=null;
      if(mid>st){l=getSubTreeRoot(st,mid-1);}
      if(ed>mid){r=getSubTreeRoot(mid+1,ed);}
      TreeNode node=new TreeNode(mnums[mid],l,r);
      return node;
  }
  
  
  public TreeNode sortedArrayToBST(int[] nums) {
      mnums = new int[nums.length];
      for(int i=0; i<nums.length; i++){
          mnums[i]=nums[i];
      }
      TreeNode ans=getSubTreeRoot(0,nums.length-1);
      return ans;
  }
}